### ISSUE TITLE
NaN on arm with fp16 but good on x86_64 linux

### ISSUE THREAD

**Issue by oewill** [Tencent/ncnn#5313]
## detail | 详细描述 | 詳細な説明

model consists of ['ExpandDims', 'Split', 'Convolution1D', 'Permute', 'UnaryOp', 'BinaryOp', 'MemoryData', 'MatMul', 'Clip', 'Reduction', 'Reshape', 'Gemm', 'Concat', 'Slice', 'Softmax', 'LayerNorm', 'GELU', 'InnerProduct']

On x86_64 linux, ncnn with fp16 works perfectly. but on arm, final results show NaN.
When i set the option, `net.opt.use_fp16_storage = false`, then it works fine, but it needs extra latency.

As other option, I test onnx-runtime on arm, it runs as fast as ncnn with fp16, and results are correct.

Just onnx-runtime optimizes the model better or something else?
I wanna make ncnn work with fp16 but don't know how to debug. 
Any suggestion makes me great help. Do you have any idea?

Thank you for your help.

**Comment by nihui** [comment]
Hi, please provide the problematic model files (param and bin)

**Comment by nihui** [comment]
针对onnx模型转换的各种问题，推荐使用最新的pnnx工具转换到ncnn
In view of various problems in onnx model conversion, it is recommended to use the latest pnnx tool to convert your model to ncnn

```shell
pip install pnnx
pnnx model.onnx inputshape=[1,3,224,224]
```

详细参考文档
Detailed reference documentation
https://github.com/pnnx/pnnx
https://github.com/Tencent/ncnn/wiki/use-ncnn-with-pytorch-or-onnx#how-to-use-pnnx
